cdap-tms/src/main/java/io/cdap/cdap/messaging/data/DefaultRawMessage.java

```diff
@@ -16,29 +16,21 @@
 
 package io.cdap.cdap.messaging.data;
 
-/**
- * Represents a unique message in the messaging system. It contains the message id and the payload
- */
-public class RawMessage {
-
+public class DefaultRawMessage implements RawMessage {
```

Comment:
Nit: We may eventually need this to be in SPI and this may be a case where a POJO should be part of SPI since it's the one extension would need to create. Otherwise extension would need to have a copy of the very same class

Reply:
I wish our java version had records :)

Reply:
Another option is to introduce a builder and skip an interface. Builders allow to safely evolve

Reply:
Agree to keep it at SPI level. Let me take a stab at it with a builder.